pkg/sql/opt_exec_factory.go

```diff
@@ -609,9 +609,18 @@ func (ef *execFactory) ConstructIndexJoin(
 		return nil, err
 	}
 
-	tableScan.index = tabDesc.GetPrimaryIndex()
+	idx := tabDesc.GetPrimaryIndex()
+	tableScan.index = idx
```

Comment:
Quick question if anyone knows the answer: what is reasoning for building the scan node directly at each node instead of using `ConstructScan`?